src/elm/Page/Dashboard/Analysis.elm

```diff
@@ -498,29 +574,105 @@ update msg model loggedIn =
+                        |> UR.addCmd (fetchAnalysis loggedIn model.filters cursor)
+
+                _ ->
+                    UR.init model
+
+        ClearSelectSelection ->
+            case model.status of
+                Loaded _ _ ->
+                    let
+                        oldFilters =
+                            model.filters
+
+                        newModel =
+                            { model
+                                | status = Loading
+                                , filters = { oldFilters | profile = Nothing }
+                                , reloadOnNextQuery = True
+                            }
+                    in
+                    newModel
+                        |> UR.init
+                        |> UR.addCmd (fetchAnalysis loggedIn newModel.filters Nothing)
 
                 _ ->
                     UR.init model
 
+        SelectStatusFilter statusFilter ->
+            let
+                oldFilters =
+                    model.filters
 
-fetchAnalysis : Shared -> Symbol -> Eos.Name -> Maybe String -> Cmd Msg
-fetchAnalysis shared communityId account maybeCursorAfter =
+                newModel =
+                    { model
+                        | filters = { oldFilters | statusFilter = statusFilter }
+                        , status = Loading
+                        , reloadOnNextQuery = True
+                    }
+            in
+            newModel
+                |> UR.init
+                |> UR.addCmd (fetchAnalysis loggedIn newModel.filters Nothing)
+
+
+fetchAnalysis : LoggedIn.Model -> Filter -> Maybe String -> Cmd Msg
+fetchAnalysis { accountName, selectedCommunity, shared } { profile, statusFilter } maybeCursorAfter =
     let
+        filterRecord =
+            { claimer = Absent
+            , status = Absent
+            }
+                |> (\fr ->
+                        { fr
+                            | claimer =
+                                case profile of
+                                    Just p ->
+                                        Present (Eos.nameToString p.account)
+
+                                    Nothing ->
+                                        Absent
+                        }
+                   )
+                |> (\fr ->
+                        { fr
+                            | status =
+                                case statusFilter of
+                                    All ->
+                                        Absent
+
+                                    Approved ->
+                                        Present "approved"
+
+                                    Rejected ->
+                                        Present "rejected"
+
+                                    Pending ->
+                                        Present "pending"
+                        }
+                   )
```

Comment:
I think anonymous functions bring a bit of confusion here. Maybe something like this would be clearer?

```elm
let
    optionalClaimer =
        case profile of
            Just p ->
                Present (Eos.nameToString p.account)

            Nothing ->
                Absent

    optionalStatus =
        case statusFilter of
            All ->
                Absent

            Approved ->
                Present "approved"

            Rejected ->
                Present "rejected"

            Pending ->
                Present "pending"

    filterRecord =
        { claimer = optionalClaimer
        , status = optionalStatus
        }
```

Reply:
wow much much better, I guess I was tired, hopefully I can count on your review bro, thanks a lot!!!